apache/tests/test_apache.py

```diff
@@ -140,3 +145,35 @@ def test_invalid_version(check):
     check._submit_metadata("invalid_version")
 
     check.log.info.assert_called_once_with("Cannot parse the complete Apache version from %s.", "invalid_version")
+
+
+@pytest.mark.parametrize(
+    'version, pattern, expected_parts',
+    [
+        ('Apache/2.4.2 (Unix) PHP/4.2.2 MyMod/1.2', VERSION_REGEX, {'major': '2', 'minor': '4', 'patch': '2'}),
+        (
+            'Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips',
+            VERSION_REGEX,
+            {'major': '2', 'minor': '4', 'patch': '6'},
+        ),
+        ('Apache/2.14.27', VERSION_REGEX, {'major': '2', 'minor': '14', 'patch': '27'}),
+    ],
+    ids=['unix_full_version', 'redhat_version', 'min_version'],
```

Comment:
Alternative-tip-just-in-case: we can put the `id` right on each parameter set using…

```python
@pytest.mark.parametrize(
    'version, expected_parts',
    [
        pytest.param(arg1, arg2, id='unix_full_version'),
        ...
    ]
)
```

which *might* be more readable.